### DISCUSSION TITLE
1v1 Versus Mode | Continuation/Expansion

### DISCUSSION THREAD

**Discussion by WalmartWhiteCat** [ppy/osu#31127] [Ideas]
A separate Multiplayer mode that pits 2 players in a head-to-head match. This mode will be based on each player's average skill level and will be fully automated to avoid one player having an unfair advantage.

To automate this, there will be a monthly scan done on every player that actively participates in the 1v1 Versus mode (played within the past month). This scan will look at the entire top 100 of the player to find their average star range, average map length and more. It will also look at more general stats such as playtime, play count, accuracy and so on.

But to keep match-ups more even, the scan will also look at the 3 most common map tags in the player's top 100 and recent scores. These tags will not be the ones chosen by mappers, but instead the new ones decided upon by the players themselves that denotes whether a map is mostly tech, aim, finger control, streams and so on.

Finally, once a player joins the queue, this gathered information will decide which players they can be matched up against. The way in which the decision is made will be if the two players' profiles are 70% similar. In other words, if 70% of the information on each profile matches up to an acceptable degree, the players will be chosen and the match will proceed.

---

Maps will also be chosen based on the information gathered by scans. The biggest factor will be average star range, map length and the 3 most common map tags between the players. A map must fit the average between the two players.

Example 1:

Player 1: Average difficulty 5*. Average 180 second map length. Common tags are tech, finger control and streams.
Player 2: Average difficulty 6*. Average 240 second map length. Common tags are jump, tech and finger control.
Result: Map will lean toward 5.5* with a preference for 210 seconds in length. Tags must include tech or finger control.

Example 2:

Player 1: Average difficulty 7.4*. Average 120 second map length. Common tags are jump, aim control and speed.
Player 2: Average difficulty 6.9*. Average 180 second map length. Common tags are tech, streams and speed.
Result: Map will lean toward 7.15* with a preference for 150 seconds in length. Tags must include speed.

Finally, I think incorporating a new way of winning would be very cool. Specifically, the way Geoguessr does it would be nice with slight deviations.

- Each player starts with 1000 health
- For each 1000 score a player gets on a map, they will do an extra 1 point of damage.
- Damage can be blocked with damage. 750k score met with 750k score will deal no damage. 750k score met with 500k score will deal 250 damage.
- After 3 maps there will be a healing round, where damage dealt is swapped with health gained. Opponent score will not factor into health gained. Healing rounds can only happen once, to avoid matches taking too long.

---

Finally, I believe there should be 2 ways of playing 1v1 Versus. First is what I've been talking about so far. Second is a casual mode:

- Profile scans will not factor into matchmaking.
- Healing rounds will become a vote instead of being automated.
- Healing rounds can still only happen once, but the option to vote after the 3rd map will stick around after every subsequent map until the vote succeeds.
- Map picks will not be automated, instead being up to the players. However, before joining a 1v1, players will choose their desired star range and then be put into the queue. Then, they will only be queued with others who's range is reasonably similar.

Example 1:

Player 1: Range set to 2* - 4*
Player 2: Range set to 2* - 4.8*
Result: These players will be queued together.

Example 2:

Player 1: Range set to 5.4* - 5.8*
Player 2: Range set to 5* - 7*
Result: These players will NOT be queued together.

After the match begins, players will pick up to 3 maps each within the chosen star range (similar to a regular multiplayer lobby) and then vote on which ones will go through (They will also vote on a healing map). This will repeat until 3 maps from each player all go through (Along with the healing map).
If one player is not knocked out by the end of all the maps, this voting process repeats. The healing map, however, will remain the same unless it was played during the match.

---

Small updates/notes:
Rank does NOT factor into matchmaking
PP also does NOT factor into matchmaking
There will be no rankings dedicated to this mode (This is why I call it 1v1 Versus as opposed to Ranked).
This mode is meant to be a new, fun way to experience your favourite rhythm game. It is not meant to be a replacement or alternative to the current ranking system.
All modes should be supported, either at initial launch or soon after initial launch.

**Comment by peppy** [comment]
cc @smoogipoo for potential ideas to add to the thought pile

**Reply by WalmartWhiteCat** [reply]
Can't wait to see what comes of this

**Comment by ziv-vy** [comment]
I'll bump this due to "Quick play". I think that, for a good part of the community, 8 player battle royale was not what they were hoping for, and having this be the competitive mode would sound nice.

**Reply by smoogipoo** [reply]
Fwiw I'm very aware of what the community wants/was expecting. It was a conscious decision not to do this for the time being to make it more of a social experience than a competitive one.

Rest assured that this isn't forgotten.

**Reply by Tenexxt** [reply]
I think this could be one of the variations of `Quick play`, you could give that multiple modes and for those that only like one specific one you could add a dropdown to choose the preferred submode to queue to.

would make QP less repetetive in the long run and would still be a fun, cassual experience.

I don't think the QP format is ready for having a competitive version, I already saw complaints about the mappool and the general flow, and since this would be a 'competetive mode' its kinda important to figure out the proper format first.

(Also we alraedy have literal ranks for like every important metric in the game and beatmap spotlights tiers so i just don't see the point lol)
